lib/ansible/modules/dnf.py

```diff
@@ -1079,9 +1081,13 @@ def ensure(self):
                         response['results'].append("Packages providing %s not installed due to update_only specified" % spec)
                 else:
                     for pkg_spec in pkg_specs:
-                        # best effort causes to install the latest package
-                        # even if not previously installed
-                        self.base.conf.best = True
+                        # Previously we forced base.conf.best=True here.
+                        # However in 2.11+ there is a self.nobest option, so defer to that.
+                        # Note, however, that just because nobest isn't set, doesn't mean that
+                        # base.conf.best is actually true. We only force it false in
+                        # _configure_base(), we never set it to true, and it can default to false.
+                        # Thus, we still need to explicitly set it here.
+                        self.base.conf.best = not self.nobest
```

Comment:
Can this change be done in a separate PR? It looks like this change makes it so this bugfix cannot be backported.

Reply:
:+1: